Framework/LegacyNexus/src/nxstack.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+  This is some code to handle a stack of NeXus files. This is used to implement
+  external linking within the NeXus-API
+
+  Copyright (C) 1997-2006 Mark Koennecke
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+  For further information, see <http://www.nexusformat.org>
+
+  Added code to support the path stack for NXgetpath,
+        Mark Koennecke, October 2009
+*/
+#include "MantidLegacyNexus/nxstack.h"
+#include "MantidLegacyNexus/napi.h"
+#include "MantidLegacyNexus/napi_internal.h"
+#include <stdlib.h>
+#include <string.h>
+
+/*-----------------------------------------------------------------------
+ Data definitions
+---------------------------------------------------------------------*/
+
+typedef struct {
+  pHDF4Function pDriver;
+  NXlink closeID;
+  char filename[1024];
+} fileHDF4StackEntry;
+
+typedef struct __fileHDF4Stack {
+  int fileStackPointer;
+  fileHDF4StackEntry fileStack[MAXEXTERNALDEPTH];
+  int pathPointer;
+  char pathStack[NXMAXSTACK][NX_MAXNAMELEN];
+} fileHDF4Stack;
```

Comment:
In `nxstack` we refer to `HDF4` in the naming of objects rather than `LegacyNexus`, is this intended or a hangover from a previous iteration of this change?